### PR TITLE
Release calendar

### PR DESCRIPTION
**Pull request by ericnatejones**
Year calendar for release dates.

I wasn't able to figure out how to utilize the calendar package's CSS, so I copied the whole thing to the component file. It felt bad. 

DOM manipulation for the tooltip might also have a better solution.

### CODE REVIEW

File: src/app/views/river-detail/components/flow-tab/flow-tab.vue [hunk 1]
```diff
@@ -438,6 +448,12 @@ export default {
     },
     handleOpenGageModal () {
       this.$refs.gageLinkModal.open()
+    },
+    calendarView () {
+      this.releaseView = "calendar"
```
Comment (ryangroth5) [inline src/app/views/river-detail/components/flow-tab/flow-tab.vue:453]: eric consider enums or consts something like : 
`const FLOWVIEW_CALENDAR=1; const FLOWVIEW_TABLE=2`
or
`const FLOWVIEWENUM = Object.Freeze({CALENDAR:1, TABLE:2});`

then assign releaseView to FLOWVIEWENUM.CALENDAR rathern than the string. 

strings have too many gotchas and they don't constrain to the limits of the number of views you want to represent.

File: src/app/views/river-detail/components/flow-tab/components/releases-calendar.vue [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+      while (e.element.childNodes.length > 1) {
+        e.element.removeChild(e.element.lastChild);
+      }
+    }
+  },
+
+};
+
+</script>
+<style lang="scss">
```
Comment (ryangroth5) [inline src/app/views/river-detail/components/flow-tab/components/releases-calendar.vue:102]: you did this right, css should be constrained to the components it affects

File: src/app/store/modules/river-events.js [hunk 3]
```diff
@@ -111,5 +120,29 @@ export default {
 
       return [];
     },
+    calendarData: (state, getters) => {
+      if(getters.releaseDates){
```
Comment (ryangroth5) [inline src/app/store/modules/river-events.js:124]: better here too.

**Comment by ericnatejones** [comment]
Thanks so much! I'll get on these and submit another PR

**Comment by ngottlieb** [comment]
@ericnatejones to import the CSS I think you have two options.

First, dig into node_modules/v-year-calendar or wherever the package is and look for the CSS file you are trying to pull in. note the path. (probably something like `v-year-calendar/dist/styles.css`). Then either:

1) import the file in the <script> section like you import other stuff ([example](https://github.com/AmericanWhitewater/wh2o-vue/blob/development/src/app/views/river-detail/components/geometry-edit-modal/components/geometry-editor.vue))

    import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css';

2) import the file in the <style> section with SASS ([example](https://github.com/AmericanWhitewater/wh2o-vue/blob/development/src/app/global/components/content-editor/content-editor.vue))

    @import "~tinymce/skins/ui/oxide/skin.min.css";

**Pull request closed without merging**